# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/wasm/spectre-mask.js
# Commit: 3ecda60613c5
# Full Hash: 3ecda60613c5cb7a57cc7004919c7d12c3a29268
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-22 03:08:27
# Description:
#   Bug 1736531 - Fix. r=jseward
#   
#   See bug for analysis.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D129010
# ==============================================================================

diff -r 1044b04cf89b -r 3ecda60613c5 js/src/jit-test/tests/wasm/spectre-mask.js
--- a/js/src/jit-test/tests/wasm/spectre-mask.js	Thu Oct 21 15:44:59 2021 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,40 +0,0 @@
-// |jit-test| skip-if: !hasDisassembler() || !(wasmCompileMode() == "baseline" || wasmCompileMode() == "ion") || !(getBuildConfiguration().x64 && !getBuildConfiguration()["arm64-simulator"] && !getBuildConfiguration()["mips64-simulator"])
-
-// What we can't test here is the direct-call-from-JIT path, as the generated
-// code is not available to wasmDis.
-
-var ins = wasmEvalText(`
-(module
-  (import "" "wasm2import" (func $g (result i32)))
-  (memory 1)
-  (type $ty (func (result i32)))
-  (table $t 1 1 funcref)
-  (func $f (result i32)
-    (i32.const 37))
-  (func (export "wasm2wasm") (result i32)
-    (call $f))
-  (func (export "wasm2import") (result i32)
-    (call $g))
-  (func (export "wasmIndirect") (result i32)
-    (call_indirect (type $ty) $t (i32.const 0)))
-  (func (export "instanceCall") (result i32)
-    (memory.size))
-)`, {'':{'wasm2import': function() {}}});
-
-switch (wasmCompileMode()) {
-case "ion":
-    assertEq(wasmDis(ins.exports.wasm2wasm, {tier:'stable', asString:true}).match(/call.*\n.*mov %eax, %eax/).length, 1);
-    assertEq(wasmDis(ins.exports.wasm2import, {tier:'stable', asString:true}).match(/call.*\n(?:.*movq.*\n)*.*mov %eax, %eax/).length, 1);
-    assertEq(wasmDis(ins.exports.wasmIndirect, {tier:'stable', asString:true}).match(/call.*\n(?:.*movq.*\n)*.*mov %eax, %eax/).length, 1);
-    assertEq(wasmDis(ins.exports.instanceCall, {tier:'stable', asString:true}).match(/call.*\n(?:.*movq.*\n)*.*mov %eax, %eax/).length, 1);
-    break;
-case "baseline":
-    assertEq(wasmDis(ins.exports.wasm2wasm, {tier:'stable', asString:true}).match(/call.*\n.*add.*%rsp\n.*mov %eax, %eax/).length, 1);
-    assertEq(wasmDis(ins.exports.wasm2import, {tier:'stable', asString:true}).match(/call.*\n.*add.*%rsp\n(?:.*movq.*\n)*.*mov %eax, %eax/).length, 1);
-    assertEq(wasmDis(ins.exports.wasmIndirect, {tier:'stable', asString:true}).match(/call.*\n.*add.*%rsp\n(?:.*movq.*\n)*.*mov %eax, %eax/).length, 1);
-    assertEq(wasmDis(ins.exports.instanceCall, {tier:'stable', asString:true}).match(/call.*\n.*add.*%rsp\n(?:.*movq.*\n)*.*mov %eax, %eax/).length, 1);
-    break;
-default:
-    throw "Unexpected compile mode";
-}
-